pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -241,6 +241,12 @@ func (tm *TableManager) cleanupCache() error {
 
 // ensureQueryReadiness compares tables required for being query ready with the tables we already have and downloads the missing ones.
 func (tm *TableManager) ensureQueryReadiness(ctx context.Context) error {
+	start := time.Now()
+	defer func() {
+		duration := time.Since(start)
+		tm.metrics.ensureQueryReadinessDurationSeconds.Observe(duration.Seconds())
```

Comment:
Since duration is not reused, you can make it concise.
```suggestion
		tm.metrics.ensureQueryReadinessDurationSeconds.Observe(time.Since(start).Seconds())
```